src/syscheckd/whodata/audit_parse.c

```diff
@@ -230,34 +236,110 @@ void clean_regex() {
     freed = 1;
 }
 
+/**
+ * @brief Looks for a specific field in an audit event.
+ *
+ * @param buffer Audit message.
+ * @param key Audit field to look for
+ * @return Value of the field specified in key.
+ */
+STATIC char *get_audit_field(const char *buffer, const char *key) {
+    char *value = NULL;
+    char *start = NULL;
+    char *ascii_value = NULL;
+    int is_hex_buffer = 1;
+    int end = 0;
```

Comment:
Since `start` references a point in the string but `end` is just an integer it could be a little misleading, I think we should rename `end` to something in the lines of `length`.